tests/system/framework/misc/ssh.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import shlex
+from functools import wraps
+from time import sleep
+from typing import Any, Callable, ParamSpec
+
+from pytest_mh.conn import Connection, Process, ProcessError, ProcessLogLevel, ProcessResult
+
+from . import to_list_of_strings
+
+
+class SSHKillableProcess(object):
+    """
+    Run an asynchronous process that requires ``SIGTERM`` to be terminated.
+    """
+
+    def __init__(
+        self,
+        client: Connection,
+        argv: list[Any],
+        *,
+        cwd: str | None = None,
+        env: dict[str, Any] | None = None,
+        input: str | None = None,
+        log_level: ProcessLogLevel = ProcessLogLevel.Full,
+    ) -> None:
+        """
+        :param client: SSH client.
+        :type client: Connection
+        :param argv: Command to run.
+        :type argv: list[Any]
+        :param cwd: Working directory, defaults to None (= do not change)
+        :type cwd: str | None, optional
+        :param env: Additional environment variables, defaults to None
+        :type env: dict[str, Any] | None, optional
+        :param input: Content of standard input, defaults to None
+        :type input: str | None, optional
+        :param log_level: Log level, defaults to ProcessLogLevel.Full
+        :type log_level: ProcessLogLevel, optional
+        """
+        if env is None:
+            env = {}
+
+        argv = to_list_of_strings(argv)
+        command = shlex.join(argv)
+        pidfile = "/tmp/.mh.sshkillableprocess.pid"
```

Comment:
This feels racy.  Not sure how we can do better though.  (Racy as in - if two tests happen to try an ssh killable process at the same time)

Reply:
Tests run sequentially and this is a hard requirement for the framework, so two tests can run at the same time